crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -0,0 +1,327 @@
+    fn message(&self) -> String {
+        let DocstringMissingException { id } = self;
+        format!("Raised exception `{id}` missing from docstring")
+    }
+}
+
+/// ## What it does'
+/// Checks for function docstrings that include exceptions which are not
+/// explicitly raised.
+///
+/// ## Why is this bad?
+/// Some conventions prefer non-explicit exceptions be left out of the docstring.
+///
+/// ## Example
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///
+///     Raises:
+///         ZeroDivisionError: Divided by zero.
+///     """
+///     return distance / time
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ```
+#[violation]
+pub struct DocstringExtraneousException {
+    ids: Vec<String>,
+}
+
+impl Violation for DocstringExtraneousException {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DocstringExtraneousException { ids } = self;
+        format!("{} not explicitly raised.", ids.join(", "))
+    }
+}
+
+// Parse docstring
+#[derive(Debug)]
+struct DocstringEntries {
+    raised_exceptions: Vec<String>,
+    raised_exceptions_range: Option<TextRange>,
+}
+
+impl DocstringEntries {
+    fn new(sections: &SectionContexts, style: SectionStyle) -> Self {
+        let mut raised_exceptions: Vec<String> = Vec::new();
+        let mut raised_exceptions_range = None;
+
+        for section in sections.iter() {
+            if section.kind() == SectionKind::Raises {
+                raised_exceptions = parse_entries(section.following_lines_str(), style);
+                raised_exceptions_range = Some(section.range());
+            }
+        }
+
+        Self {
+            raised_exceptions,
+            raised_exceptions_range,
+        }
+    }
+}
+
+fn parse_entries(content: &str, style: SectionStyle) -> Vec<String> {
+    match style {
+        SectionStyle::Google => parse_entries_google(content),
+        SectionStyle::Numpy => parse_entries_numpy(content),
+    }
+}
+
+fn parse_entries_google(content: &str) -> Vec<String> {
+    let mut entries: Vec<String> = Vec::new();
+    for potential in content.split('\n') {
+        let Some(colon_idx) = potential.find(':') else {
+            continue;
+        };
+        let entry = potential[..colon_idx].trim().to_string();
+        entries.push(entry);
+    }
+    entries
+}
+
+fn parse_entries_numpy(content: &str) -> Vec<String> {
```

Comment:
Can you document this and the above to give an example of the expected structure that it's parsing?